src/com/perl5/lang/perl/util/PerlRunUtil.java

```diff
@@ -83,6 +100,82 @@ public static GeneralCommandLine getPerlCommandLine(@NotNull Project project, @N
 		return commandLine;
 	}
 
+	private static List<String> getPerlLibraries(@NotNull Project project, PerlConfiguration runProfile, @Nullable VirtualFile scriptFile)
+	{
+		if (PlatformUtils.isIntelliJ())
```

Comment:
We may return immediately if profile state is null or not intellij. Try to avoid code ladders